torch/csrc/jit/serialization/pickler.cpp

```diff
@@ -353,6 +353,41 @@ void Pickler::pushTensor(const IValue& ivalue) {
   }
 }
 
+void Pickler::pushLiteralSparseTensor(const at::Tensor& tensor) {
+  TORCH_INTERNAL_ASSERT(tensor.is_sparse());
+  // The arguments to this function are:
+  // size, requires_grad, pinned memory, data_type, device,
+  // indices, values, backward_hooks
+  pushGlobal("torch._utils", "_rebuild_sparse_coo_tensor");
+  // size
+  push<PickleOpCode>(PickleOpCode::MARK);
+  push<PickleOpCode>(PickleOpCode::MARK);
+  for (auto size : tensor.sizes()) {
+    pushInt(size);
+  }
+  push<PickleOpCode>(PickleOpCode::TUPLE);
+  // requires_grad
+  pushIValue(tensor.requires_grad());
+  // pinned memory
+  pushBool(tensor.options().pinned_memory());
+  // data type
+  pushInt(static_cast<uint16_t>(tensor.scalar_type()));
+  // device
+  pushString(tensor.device().str());
```

Comment:
the PR description mentioned we are adding CPU sparse tensor support, but here it looks like we are adding a generic pickle/unpickle support for tensor (i.e. in all devices), is this guarded somewhere in RPC side to ensure we are only sending cpu sparse tensor?

Reply:
No, I did not add any guards. The next pr in the stack adds necessary changes for tensors on a CUDA device. 

A guard will cause the application to fail. So I don't understand why we would need a guard because the application will also fail without the guard.